cmd/k8s-operator/deploy/chart/Chart.yaml

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) Tailscale Inc & AUTHORS
+# SPDX-License-Identifier: BSD-3-Clause
+
+apiVersion: v1
+name: tailscale-operator
+description: A Helm chart for Tailscale Kubernetes operator
+
+# A chart can be either an 'application' or a 'library' chart.
+#
+# Application charts are a collection of templates that can be packaged into versioned archives
+# to be deployed.
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "unstable"
```

Comment:
@irbekrm @mayakacz @rhealitycheck  any thoughts on how we version these?
Should we just follow the tailscale versioning?

Reply:
@maisem 
https://helm.sh/docs/topics/charts/
> version: A SemVer 2 version (required)
> appVersion: The version of the app that this contains (optional). Needn't be SemVer. Quotes recommended.

`appVersion` should likely match the latest stable operator version, so `v1.50.1` for the container now. If this changes, bump the patch component of `version`

`version` is SemVer for the *chart itself*. i.e., bump the main version if you're breaking backward compatibility. Bump minor/patch when making any change to the chart. 0.1.0 would be appropriate for the first merge (0.x.y indicating that it's not really done). This is independent of `appVersion`. For example, if the entire chart got refactored, that's probably a major version change, but it wouldn't be a tailscale major version. Conversely, there might not be any structural changes to the chart for a very long time, making it forward compatible with tailscale `v3.1.2`.

Here's a couple of examples:
- https://github.com/grafana/helm-charts/pull/2716/files (chart added a field, patch++)
- https://github.com/grafana/helm-charts/pull/2706/files (appVersion bump, patch++)

Reply:
> Should we just follow the tailscale versioning?

Yes, previously I have bumped this every time I release a new chart version to make sure the right images get pulled.
This is also the tag of the operator and proxy images so I've set this to `unstable` here because that's what we currently have in the plain manifests.

It doesn't really matter what versions are there _in this PR_ because we aren't actually going to publish with these versions.Once we set up infra to actually publish charts we'll have to have a discussion about publishing flow and versioning. And before we publish we'll have to bump these anyway.

Reply:
wrt, to the chart version, I am not sure whether we are likely to be publishing new chart versions outside of Tailscale release flow. If we do not, then having the same app and chart version is fine. We did that for cert-manager for example.

Reply:
Actually, thinking about it, we shouldn't need to ever bump it here. It should be ok that both `appVersion` and `version` are dummy values and we set them at publish time `helm package <path> --app-version v0.52.0 --version v0.52.0`.
See i.e https://github.com/cert-manager/cert-manager/blob/master/deploy/charts/cert-manager/Chart.template.yaml#L3-L5
I've set the proxy tag in values.yaml to `""` to ensure that `unstable` does not override `appVersion` that we default to.